client/state/data-getters/index.js

```diff
@@ -100,3 +100,32 @@ export const requestSiteAlerts = siteId => {
 		}
 	);
 };
+
+export const requestGutenbergDraftPost = siteId =>
+	requestHttpData(
+		`gutenberg-draft-post-${ siteId }`,
+		http(
+			{
+				path: `/sites/${ siteId }/posts/create-draft`,
+				method: 'POST',
+				apiNamespace: 'wp/v2',
+				body: {},
+			},
+			{}
+		),
+		{ formApi: () => ( { ID } ) => [ [ `gutenberg-draft-post-${ siteId }`, ID ] ] }
+	);
```

Comment:
We can probably remove this for now and revisit how to handle autosave drafts in a follow up PR. `requestHttpData` here feels incorrect for this POST case and is even more unexpected in `mapStateToProps`. Overall I think we'd want to trigger specific requests to create the draft only after we have some user input.

If we want to keep this pretty lightweight later I think we'd end up making a create action, but we could potentially still keep a poll going of the post object using requestHttpData.

Thoughts here @dmsnell ?